### ISSUE TITLE
network parameters should be < number of data points/N

### ISSUE THREAD

**Issue by torrmal** [mindsdb/lightwood#72]
lets start working on making sure taht we dont have too many hyper parameteres, and that the neural network is not simply memorizing but finding a minima, to that light we can try to make it by having a parameter where we define the threshold of parameters in the neural network compared to the number of data points

https://bair.berkeley.edu/blog/2019/08/13/memorization/

**Comment by George3d6** [comment]
@paxcema can you take a look at this at some point, seems like an important point that we completely forgot about.

**Comment by paxcema** [comment]
After discussing with @torrmal, we've decided to focus our efforts on the current dropout and early stopping strategies, as 1) the memorization problem is correlated with overfitting in most domains and 2) research suggests that overparametrization, when regularized, is beneficial. We'll revisit memorization as defined in the Carlini paper when text generation is a more prominent use case in MindsDB, which is when privacy concerns might arise.

**Comment by paxcema** [comment]
Dropout has been addressed in #325, and early stopping seems to be working fine, so I'll close this (at least until we have unconstrained language generation as a valid use case).
